network/server_test.go

```diff
@@ -690,19 +698,48 @@ func TestMinimumBootNodeCount(t *testing.T) {
 			shouldFail: false,
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, createErr := CreateServer(&CreateServerParams{
+			srv, createErr := CreateServer(&CreateServerParams{
 				ConfigCallback: func(c *Config) {
 					c.Chain.Bootnodes = tt.bootNodes
+					c.NoDiscover = false
 				},
 			})
-
 			if tt.shouldFail {
 				assert.Error(t, createErr)
 			} else {
 				assert.NoError(t, createErr)
+				assert.NoError(t, srv.Close())
 			}
 		})
 	}
 }
+
+func TestTemporaryDial(t *testing.T) {
+	defaultConfig := &CreateServerParams{
+		ConfigCallback: func(c *Config) {
+			c.NoDiscover = true
+		},
+	}
+	servers, err := createServers(2, map[int]*CreateServerParams{0: defaultConfig, 1: defaultConfig})
+
+	assert.NoError(t, err)
+
+	t.Cleanup(func() {
+		closeTestServers(t, servers)
+	})
+	//add server1 to server0 temperary dial list
+	servers[0].temporaryDials.Store(servers[1].host.ID(), true)
+	//connect server0 to server1
+	err = servers[0].host.Connect(context.Background(), *servers[1].AddrInfo())
+	assert.NoError(t, err)
+	// ensure that the connection is established
+	assert.Equal(t, network.Connected, servers[0].host.Network().Connectedness(servers[1].host.ID()))
+	// wait until identity handshake is done
+	time.Sleep(10 * time.Second)
+	// since it is temporary dial, server should not contain its peer in the peerlist
+	assert.NotContains(t, servers[0].peers, servers[1].host.ID())
+	assert.NotContains(t, servers[1].peers, servers[0].host.ID())
```

Comment:
Please clean up the test so it's more readable

Reply:
Done,Commit-[link](https://github.com/0xPolygon/polygon-edge/pull/343/commits/ea9b338d7b7cced2866e0ee80785c2ccd4fdf7f2)